cpp/tests/binaryop/binop-integration-test.cpp

```diff
@@ -1,2716 +0,0 @@
-  auto const expected = fp_wrapper<RepType>{{142, 285, 428}, scale_type{1}};
-
-  auto const type   = data_type{type_to_id<decimalXX>(), 1};
-  auto const result = cudf::binary_operation(lhs, rhs, cudf::binary_operator::DIV, type);
-
-  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, result->view());
-}
-
-TYPED_TEST(FixedPointTestAllReps, FixedPointBinaryOpThrows)
-{
-  using namespace numeric;
-  using decimalXX = TypeParam;
-  using RepType   = device_storage_type_t<decimalXX>;
-
-  auto const col           = fp_wrapper<RepType>{{100, 300, 500, 700}, scale_type{-2}};
-  auto const non_bool_type = data_type{type_to_id<decimalXX>(), -2};
-  EXPECT_THROW(cudf::binary_operation(col, col, cudf::binary_operator::LESS, non_bool_type),
-               cudf::logic_error);
-}
-
-template <typename T>
-struct FixedPointTest_64_128_Reps : public cudf::test::BaseFixture {
-};
-
-using Decimal64And128Types = cudf::test::Types<numeric::decimal64, numeric::decimal128>;
-TYPED_TEST_SUITE(FixedPointTest_64_128_Reps, Decimal64And128Types);
-
-TYPED_TEST(FixedPointTest_64_128_Reps, FixedPoint_64_128_ComparisonTests)
-{
-  using namespace numeric;
-  using decimalXX = TypeParam;
-  using RepType   = device_storage_type_t<decimalXX>;
-
-  for (auto const rhs_value : {10000000000000000, 100000000000000000}) {
-    auto const lhs       = fp_wrapper<RepType>{{33041, 97290, 36438, 25379, 48473}, scale_type{2}};
-    auto const rhs       = make_fixed_point_scalar<decimalXX>(rhs_value, scale_type{0});
-    auto const trues     = wrapper<bool>{{1, 1, 1, 1, 1}};
-    auto const falses    = wrapper<bool>{{0, 0, 0, 0, 0}};
-    auto const bool_type = cudf::data_type{type_id::BOOL8};
-
-    auto const a = cudf::binary_operation(lhs, *rhs, binary_operator::LESS, bool_type);
-    auto const b = cudf::binary_operation(lhs, *rhs, binary_operator::LESS_EQUAL, bool_type);
-    auto const c = cudf::binary_operation(lhs, *rhs, binary_operator::GREATER, bool_type);
-    auto const d = cudf::binary_operation(lhs, *rhs, binary_operator::GREATER_EQUAL, bool_type);
-    auto const e = cudf::binary_operation(*rhs, lhs, binary_operator::GREATER, bool_type);
-    auto const f = cudf::binary_operation(*rhs, lhs, binary_operator::GREATER_EQUAL, bool_type);
-    auto const g = cudf::binary_operation(*rhs, lhs, binary_operator::LESS, bool_type);
-    auto const h = cudf::binary_operation(*rhs, lhs, binary_operator::LESS_EQUAL, bool_type);
-
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(a->view(), trues);
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(b->view(), trues);
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(c->view(), falses);
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(d->view(), falses);
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(e->view(), trues);
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(f->view(), trues);
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(g->view(), falses);
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(h->view(), falses);
-  }
-}
-
-}  // namespace binop
-}  // namespace test
-}  // namespace cudf
-
-CUDF_TEST_PROGRAM_MAIN()
```

Comment:
This appears to be a very comprehensive test suite that we're deleting. Are the contents of this integration test for JIT binary operations covered by new/existing tests for compiled binary operations? (I haven't checked the scope of the tests very closely because they're 2700 lines, but I don't see an equivalent number of lines of tests for compiled binary operations unless it's just split across multiple files.)

Reply:
compiled binary ops tests for most of the relevant combinations. Each tests has a comment above describing valid combinations and the types listed below in TYPED_TEST_SUITE.

Only 2 jit tests failed in compiled binary ops. (tested before removing it in this PR).
one of them is breaking change (supporting output type of boolean operation should be bool only).
another should be an unsupported type operation (bitwise operation between float and integer types).

Reply:
Also, binops jit test code has separate tests for each type combination, while compiled binops test code is written with Typed tests. So number of lines are less because of that too.

BinaryOperationCompiledTest: 278 tests.
BinaryOperationIntegrationTest: 106 tests.

Reply:
Awesome, thanks for looking into that. Resolving.